pkg/network/vmispec/defaults.go

```diff
@@ -50,6 +50,8 @@ func SetDefaultNetworkInterface(config netClusterConfiger, spec *v1.VirtualMachi
 		spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
 	case v1.DeprecatedSlirpInterface:
 		return fmt.Errorf("slirp interface is deprecated as of v1.3")
+	case v1.DeprecatedPasstInterface:
+		return fmt.Errorf("passt interface is deprecated as of v1.3")
```

Comment:
Slirp is handled here because it was not protected using a FG.
Passt on the other hand was enabled through a FG, therefore the warning is in the FG context [1].

[1] https://github.com/kubevirt/kubevirt/blob/a3b5ee1fc5eddd4a31ec29cf6493a7834cbda5ff/pkg/virt-config/deprecation/feature-gates.go#L106

Reply:
The linter failed here with the following error message:
```
 missing cases in switch of type v1.NetworkInterfaceType: v1.DeprecatedPasstInterface (exhaustive)
        switch v1.NetworkInterfaceType(config.GetDefaultNetworkInterface()) 
```

The alternative is to silence the linter.
WDYT?

Reply:
I think the linter asks to be explicit about not treating other options.
Does adding `default: // do nothing` satisfy the linter?

Reply:
> I think the linter asks to be explicit about not treating other options. Does adding `default: // do nothing` satisfy the linter?

No:
```
pkg/network/vmispec/defaults.go:43:2: missing cases in switch of type v1.NetworkInterfaceType: v1.DeprecatedPasstInterface (exhaustive)
        switch v1.NetworkInterfaceType(config.GetDefaultNetworkInterface()) {
```

Reply:
BTW, this code snippet is responsible for adding the default pod network with the cluster-wide default binding and is called before the FG check.

Reply:
Then a noop on that case?
I fail to understand why it does not complain about macvtap then.

And it does not make a lot of sense in general. If we had 20 options, do we need to express them all? Something is wrong here.

Reply:
It examines all the options of `NetworkInterfaceType`:

Reply:
Choose one of the options:
- Add it with a noop.
- Update the linter config to supress it in case a default is defined: https://golangci-lint.run/usage/linters/#exhaustive
- Remove it from that const group and mimic the old macvtap.

Reply:
I am in favor of removing the option from the const group, I think it also makes sense to remove slirp as well (not necessarily in this PR).
WDYT?

Reply:
Fine by me.
But slirp is a special case, it got discontinued from ga with an exception. I would not touch it to express that.

Reply:
Done.